includes/wikia/models/DesignSystemGlobalFooterModel.class.php

```diff
@@ -26,6 +26,15 @@ public function __construct( $product, $productInstanceId, $lang = self::DEFAULT
 
 	public function getData() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This function "getData" has 193 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)